.github/workflows/test_on_push.yml

```
@@ -30,18 +30,14 @@ jobs:
           python -m pip install pre-commit
           pre-commit run -a
 
-  run_unit_tests:
+  run_unit_and_integration_tests:
     needs: style
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest, macos-12, windows-latest]
-        python-version: ["3.9", "3.10", "3.11", "3.12"]
-        # We check coverage on Ubuntu with Python 3.12, so we skip unit tests for it here
-        exclude:
-          - os: ubuntu-latest
-            python-version: "3.12"
+        os: [ ubuntu-latest, macos-12, windows-latest ]
```

```suggestion
        os: [ubuntu-latest, macos-12, macos-14, windows-latest]
```
Besides this change, we can remove the `include:` section and the separate entries for macOS 14 below. We kept them because Python 3.8 and Python 3.9 were not available back then: https://github.com/actions/setup-python/issues/808